### ISSUE TITLE
Setup auth(n/z)

### ISSUE THREAD

**Issue by marchchad** [marchchad/dadwind#2]
Set up authentication and authorization.

For now, let's leverage some OAuth providers (i.e., Facebook, Google, etc.). May use firebase or just manage our own users.

Configure basic roles for `user` and `admin`.

**Comment by Gruzzly-bear** [comment]
Considering that everyone has a google account over a Facebook, we should probably prioritize that first. It'll fit nicely if we decide on using the google maps API further down the road for the event planning community-esque feature. I know there are a few other map providers.

**Comment by marchchad** [comment]
You're right that it will help with other integrations down the road. Maybe Twitter over facebook to use as another oauth provider. I'd expect most people to have a Google login, but not everyone does.

Yeah I've working with a lot of them over the years. My degrees are in geography and I've worked with google maps, leaflet, esri js api, mapbox, openstreetmap, bing maps, and probably some others I'm forgetting.

**Comment by Gruzzly-bear** [comment]
That's awesome. I knew there was a bunch but the fact you have all that knowledge with this is amazing.
